### PR TITLE
Updated to support current versions of IntelliJ

### PR DESCRIPTION
**Pull request by msedgren**
I updated to work with the most recent version of IntelliJ. I also tested with other versions and it worked well. In addition, I switched to use the IntelliJ logging framework (over console logging) and fixed some possible NPEs (cleanup and processing unknown commands).

### CODE REVIEW

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<idea-plugin version="2">
+<idea-plugin>s
```
Comment (anonfunc) [inline src/main/resources/META-INF/plugin.xml]: Trailing "s"

Comment (msedgren) [reply src/main/resources/META-INF/plugin.xml]: Sorry to miss. Will fix in just a bit...

Comment (msedgren) [reply src/main/resources/META-INF/plugin.xml]: fixed

**Comment by msedgren** [comment]
I realized after submitting this pr that I should have probably started a conversation using issues first. In any case, I started using Talon a bit ago to help with my RSI. On doing so I found that I could not use it with my current version of IntelliJ Ultimate (assuming all are mostly the same) unless I installed an older version. The error I got matched https://github.com/anonfunc/intellij-voicecode/issues/13#issue-1221282717, although for me it crashed my IDE and forced me to find the plugin directory and manually delete it.

In any case, I opened the project and removed the deprecated component stuff, switching instead to using a listener. This fixed the issue for me and I have been using the manually installed version of the plugin for the last week. I also switched to JDK17 (I believe it is required on the newest versions of IntelliJ) and fixed an NPE that could occur when an unrecognized command was passed in.

I hope this helps to add some context on the change and that it will be helpful to anyone else trying to use later versions of ultimate.

**Comment by msedgren** [comment]
I noticed that the release action failed due to an invalid java source release of 17 combined with a JVM of 11. This matches my testing although I can't test fully as I am unable to publish. Anyway, I'd be happy to bump the version in a new pull request if that is alright.

**Pull request merged**